ql/experimental/callablebonds/treecallablebondengine.cpp

```diff
@@ -71,12 +76,22 @@ namespace QuantLib {
             lattice = model_->tree(timeGrid);
         }
 
+        OneFactorModel::ShortRateTree *sr=
+            dynamic_cast<OneFactorModel::ShortRateTree*>(&(*lattice));
+        if (sr)
+            sr->setSpread(s);
+        else
+            QL_REQUIRE( s==0.0,
+                        "Spread is not supported for trees other than OneFactorModel");
+
         Time redemptionTime =
             dayCounter.yearFraction(referenceDate,
                                     arguments_.redemptionDate);
         callableBond.initialize(lattice, redemptionTime);
         callableBond.rollback(0.0);
         results_.value = results_.settlementValue = callableBond.presentValue();
+        if (sr)
```

Comment:
You don't need this.  `sr` is going to be destroyed as soon as this method exits.

Reply:
I can't now recall how I traced the logic of this function. Just to check: are you sure the `lattice_`  member will never be used?

Reply:
Oh, I see.  `sr` might be an alias to an existing data member.  You're right, it should be set back.  But in this case, you'll need to make sure that the spread is reset even if an exception is raised.